package/MDAnalysis/analysis/diffusionmap.py

```diff
@@ -0,0 +1,432 @@
+        if DistanceMatrix.calculated:
+            self._dist_matrix = DistanceMatrix
+        else:
+            raise AttributeError('Distance Matrix does not exist, was'
+                                 'DistanceMatrix.run() called?')
+
+        self._nframes = self._dist_matrix.nframes
+        self._t = timescale
+
+        # check determine_epsilon called
+        if Epsilon.calculated:
+            self._kernel = Epsilon.scaled_matrix
+        else:
+            raise AttributeError('scaled_matrix does not exist, was'
+                                 'determine_epsilon() called?')
+        self._epsilon = Epsilon
+
+        if weights is None:
+            # weights do not apply to metric but density of data
+            self._weights_ker = np.ones((self._nframes,))
+        else:
+            if weights.shape[0] != self._nframes:
+                raise ValueError("The weight should have the same length as "
+                                 'the trajectory')
+            else:
+                # weights are constructed as relative to the mean
+                self._weights_ker = (np.asarray(weights, dtype=np.float64) /
+                                     np.mean(weights))
+
+    def decompose_kernel(self):
+        # this should be a reference to the same object as
+        # self.dist_matrix.dist_matrix
+        # take negative exponent of scaled matrix to create Isotropic kernel
+        self._kernel = np.exp(-self._kernel)
+
+        # define an anistropic diffusion term q
+        q_vector = np.zeros((self._nframes, ))
+        # weights should reflect the density of the points on the manifold
+        for i in range(self._nframes):
+            q_vector[i] = np.dot(self._kernel[i, :], self._weights_ker)
+
+        # Form a new kernel from the anisotropic diffusion term q
+        self._kernel /= np.sqrt(q_vector[:, np.newaxis].dot(q_vector[np.newaxis]))
+
+        # Weighted Graph Laplacian normalization on this new graph
+        d_vector = np.zeros((self._nframes, ))
+        for i in range(self._nframes):
+            d_vector[i] = np.dot(self._kernel[i, :], self._weights_ker)
+
+        for i in range(self._nframes):
+            self._kernel[i, :] = self._kernel[i, :] * self._weights_ker
+
+        # Define anisotropic transition by dividing kernel by this term
+        self._kernel /= np.sqrt(d_vector[:, np.newaxis].dot(d_vector[np.newaxis]))
+
+        # Apply timescaling
+        for i in range(self._t):
+            if i > 0:
+                self._kernel = self._kernel.dot(self._kernel)
+
+        eigenvals, eigenvectors = np.linalg.eig(self._kernel)
+
+        eg_arg = np.argsort(eigenvals)
+        self.eigenvalues = eigenvals[eg_arg[::-1]]
+        self.eigenvectors = eigenvectors[eg_arg[::-1], :]
+
+    def _prepare(self):
+        self.diffusion_space = np.zeros((self.eigenvectors.shape[0],
+                                         self.num_eigenvectors))
+
+    def _single_frame(self):
```

Comment:
this doesn't make any sense here. You are only operating on object in RAM already. The `run()` methodology should be used then we have to iterate over a universe, which we now are doing for no reason at all since you don't use the loaded timestep.